edx_notifications/lib/consumer.py

```diff
@@ -0,0 +1,101 @@
+"""
+All in-proc API endpoints for acting as a Notifications Consumer
+NOTE that we can only query for notifications that are "durable", i.e.
+persisted in our database
+"""
+
+import pytz
+from datetime import datetime
+from contracts import contract
+
+from edx_notifications.stores.store import notification_store
+
+
+@contract(user_id='int,>0')
+def get_notifications_count_for_user(user_id, filters=None):
+    """
+    Returns the number of notifications this user has
+
+    ARGS:
+        user_id: The id of the user to query
+        filters: A dict containing the following optional keys
```

Comment:
Would it be better to just have these as optional arguments to this function? Or to have a Filter class that defines the set of valid filters?